tests/request.spec.coffee

```diff
@@ -290,3 +290,64 @@ describe 'Request:', ->
 				.get('body')
 				m.chai.expect(promise).to.eventually.become(result: 'success')
 
+		describe 'given an endpoint that will time out', ->
+
+			NODE_TIMEOUT_ERROR = new Error('node timeout error')
+
+			beforeEach ->
+				@clock = m.sinon.useFakeTimers()
+
+			afterEach ->
+				@clock.restore()
+
+			stubTimeout = -> new Promise (resolve, reject) ->
+				fetchMock.get 'http://infinite-wait.com', (url, opts) ->
+					resolve()
+
+					# Emulate node-fetch timeout behaviour if we use it
+					if not IS_BROWSER and opts.timeout
+						new Promise (resolve, reject) ->
+							setTimeout((-> reject(NODE_TIMEOUT_ERROR)), opts.timeout)
+					# Browser/no timeout fetch() simply never resolves.
+					else new Promise(->)
```

Comment:
Sure, works for me, done.

I'm going to mark the Hound thing down as a minor annoyance for now and ignore it until it comes and bites me again. Definitely not perfect, but doesn't seem worth chasing up, especially if we're slowly moving away from CS anyway.